### PR TITLE
Added support to specify FSx Windows file server volumes

### PR DESCRIPTION
**Pull request by gauravkohli**
## what
* Currently we can only add `Docker/EFS` volumes to the ECS task definition. With this PR added support to add `FSx Windows file server` volumes to the ECS task definitions.

## why
* We need to deploy ECS task which runs on Windows EC2 instances and be able to access FSx volumes and with this PR it's possible to configure that now.

## references
* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ecs_task_definition#example-using-fsx_windows_file_server_volume_configuration

### CODE REVIEW

**Comment by gauravkohli** [comment]
@nitrocode/ @Gowiem / @joe-niland ... can anyone have a look at this PR.

**Review by Gowiem — APPROVED** [review]
@gauravkohli This looks solid to me -- Well done. 

Before we merge, I would like a second set of eyes on this code since it's so deeply nested. @nitrocode or @joe-niland mind putting eyes on this?

File: variables.tf [hunk 1]
```diff
@@ -279,7 +279,7 @@ variable "efs_volumes" {
     }))
   }))
 
-  description = "Task EFS volume definitions as list of configuration objects. You cannot define both Docker volumes and EFS volumes on the same task definition."
+  description = "Task EFS volume definitions as list of configuration objects. You can define only one of Docker volumes, EFS volumes and FSx volumes on the same task definition."
```
Comment (joe-niland) [inline variables.tf:282]: @gauravkohli just wanted to clarify the var description: when you say "one of" does that mean "one of each type" or "one type"? It seems mixing types is supported (just based on the AWS docs)

**Review by joe-niland — COMMENTED** [review]
@Gowiem it looks logically correct. I don't think we have an easy way of testing it with the current setup as it will require creating a Windows  ECS task.

File: variables.tf (hunk 1 above)
Comment (gauravkohli) [reply variables.tf:282]: @joe-niland .. comment change was just following the old comment which looks like meant we can only have either Docker volumes or EFS volumes on the same task definition. So I added we can only have one type, but multiple volumes of the same type. 

But then I tested it on the ECS task and actually as you said we can indeed mix types, so we can have a task definition that can have all Docker, EFS & Fsx volumes attached to it and AWS doesn't complain about it.

Comment (gauravkohli) [reply variables.tf:282]: maybe useful info to add would be that the list `fsx_windows_file_server_volume_configuration` can only have one element, as that list is used for the dynamic `volume` block, and each `volume` block can only have one of `fsx_windows_file_server_volume_configuration`

Comment (joe-niland) [reply variables.tf:282]: @gauravkohli that sounds like a good change, as well as perhaps changing 'only one of' to 'multiple'?

**Comment by joe-niland** [comment]
Thanks for your contribution @gauravkohli - released

**Pull request merged**